drivers/clock_control/clock_stm32_ll_common.c

```diff
@@ -459,6 +459,53 @@ static void set_up_plls(void)
 	}
 	LL_RCC_PLL_Disable();
 
+#if defined(STM32_PLL2_ENABLED)
+	/* Disable PLL2 after switching to HSI for SysClk and
+	 * disabling PLL since PLL source can be PLL2. */
+	LL_RCC_PLL2_Disable();
+
+	uint32_t pll_mul, pll_div;
+
+	/*
+	 * PLL2MUL on SOC_STM32F10X_CONNECTIVITY_LINE_DEVICE
+	 * 8  -> LL_RCC_PLL2_MUL_8  -> 0x00000600
+	 * 9  -> LL_RCC_PLL2_MUL_9  -> 0x00000700
+	 * ...
+	 * 14 -> LL_RCC_PLL2_MUL_14 -> 0x00000C00
+	 * 16 -> LL_RCC_PLL2_MUL_16 -> 0x00000E00
+	 * 20 -> LL_RCC_PLL2_MUL_20 -> 0x00000F00
+	 */
+	if (STM32_PLL2_MULTIPLIER == 16) {
+		pll_mul = RCC_CFGR2_PLL2MUL16;
+	} else if (STM32_PLL2_MULTIPLIER == 20) {
+		pll_mul = RCC_CFGR2_PLL2MUL20;
+	} else {
+		pll_mul = ((STM32_PLL2_MULTIPLIER - 2) << RCC_CFGR2_PLL2MUL_Pos);
+	}
+
+	/*
+	 * SOC_STM32F10X_CONNECTIVITY_LINE_DEVICE
+	 * 1  -> LL_RCC_HSE_PREDIV2_DIV_1  -> 0x00000000
+	 * 2  -> LL_RCC_HSE_PREDIV2_DIV_2  -> 0x00000010
+	 * ...
+	 * 16 -> LL_RCC_HSE_PREDIV2_DIV_16 -> 0x000000F0
+	 */
+	pll_div = ((STM32_PLL2_PREDIV - 1) << RCC_CFGR2_PREDIV2_Pos);
+
+	/* Configure PLL source */
+	if (!IS_ENABLED(STM32_PLL2_SRC_HSE)) {
+		__ASSERT(0, "Invalid source");
+	}
```

Comment:
Please move this to a `config_pll2` dedicated function that would be provided in `drivers/clock_control/clock_stm32f1.c`